trin-storage/src/versioned/id_indexed_v1/store.rs

```diff
@@ -76,26 +81,41 @@ impl IdIndexedV1Store {
         self.metrics
             .report_storage_capacity_bytes(self.config.storage_capacity_bytes as f64);
 
-        let usage_stats = self.get_usage_stats()?;
+        // query real usage stats and update "usage_stats" table to match.
+        let conn = self.config.sql_connection_pool.get()?;
+        let usage_stats = conn.query_row(
+            &sql::entry_count_and_size(&self.config.content_type),
+            [],
+            |row| {
+                let used_capacity: f64 = row.get("used_capacity")?;
+                Ok(UsageStats {
+                    entry_count: row.get("count")?,
+                    total_entry_size_bytes: used_capacity.round() as u64,
+                })
+            },
+        )?;
+        usage_stats.report_metrics(&self.metrics);
+        update_usage_stats(&conn, &self.config.content_type, &usage_stats)?;
```

Comment:
If `usage_stats` is being tracked by triggers why do we recalculate the values for the tables on each reboot? I am not saying this is bad, but wouldn't this not be needed unless there was a bug?

Souldn't we just be able to get the stats from the table on boot?

This isn't really a performance concern as this is only on boot so not a blocker just an interesting thing to think about

Reply:
It's more of a safety check, making sure it has correct value. I added `error!` if they are different.

But you are right, it shouldn't be needed in most cases.

But there might be some cases when it might be needed, e.g. after transitioning from one version to another (of course depending on how transition is done, it might not be needed in this case either).

Reply:
That is cool you added an error because now we will be aware if a bug was introduced :rocket: